src/server/game/Entities/Player/Player.cpp

```diff
@@ -30836,3 +30836,44 @@ bool Player::CanExecutePendingSpellCastRequest()
 
     return true;
 }
+
+void Player::AddMoveImpulse(Position direction)
+{
+    auto addImpulse = WorldPackets::Movement::MoveAddImpulse();
+    addImpulse.MoverGUID = GetGUID();
+    addImpulse.SequenceIndex = m_movementCounter++;
+    addImpulse.Direction = direction;
+    SendMessageToSet(addImpulse.Write(), true);
+}
+
+void Player::InitAdvancedFly()
+{
+    FlightCapabilityEntry const* flightCapabilityEntry = sFlightCapabilityStore.LookupEntry(GetFlightCapabilityID());
+    if (!flightCapabilityEntry)
+        return;
+
+    std::vector<std::tuple<OpcodeServer, float, Optional<float>>> advFlyValues = {
+        { SMSG_MOVE_SET_ADV_FLYING_AIR_FRICTION,                flightCapabilityEntry->AirFriction,                 {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_MAX_VEL,                     flightCapabilityEntry->MaxVel,                      {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_LIFT_COEFFICIENT,            flightCapabilityEntry->LiftCoefficient,             {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_DOUBLE_JUMP_VEL_MOD,         flightCapabilityEntry->DoubleJumpVelMod,            {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_GLIDE_START_MIN_HEIGHT,      flightCapabilityEntry->GlideStartMinHeight,         {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_ADD_IMPULSE_MAX_SPEED,       flightCapabilityEntry->AddImpulseMaxSpeed,          {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_BANKING_RATE,                flightCapabilityEntry->BankingRateMin,              flightCapabilityEntry->BankingRateMax               },
+        { SMSG_MOVE_SET_ADV_FLYING_PITCHING_RATE_DOWN,          flightCapabilityEntry->PitchingRateDownMin,         flightCapabilityEntry->PitchingRateDownMax          },
+        { SMSG_MOVE_SET_ADV_FLYING_PITCHING_RATE_UP,            flightCapabilityEntry->PitchingRateUpMin,           flightCapabilityEntry->PitchingRateUpMax            },
+        { SMSG_MOVE_SET_ADV_FLYING_TURN_VELOCITY_THRESHOLD,     flightCapabilityEntry->TurnVelocityThresholdMin,    flightCapabilityEntry->TurnVelocityThresholdMax     },
+        { SMSG_MOVE_SET_ADV_FLYING_SURFACE_FRICTION,            flightCapabilityEntry->SurfaceFriction,             {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_OVER_MAX_DECELERATION,       flightCapabilityEntry->OverMaxDeceleration,         {}                                                  },
+        { SMSG_MOVE_SET_ADV_FLYING_LAUNCH_SPEED_COEFFICIENT,    flightCapabilityEntry->LaunchSpeedCoefficient,      {}                                                  },
+    };
+
+    for (auto const& tuple : advFlyValues)
```

Comment:
(sidenote when using containers with tuples: you can name the fields `for (auto const& [opcode, speed, maxSpeed] : advFlyValues)`)

Reply:
Oooh, thanks !